libpam/pam_handlers.c

```diff
@@ -278,11 +279,15 @@ static int _pam_parse_conf_file(pam_handle_t *pamh, FILE *f
 	    if (res != PAM_SUCCESS) {
 		pam_syslog(pamh, LOG_ERR, "error loading %s", mod_path);
 		D(("failed to load module - aborting"));
+		free(buf);
 		return PAM_ABORT;
 	    }
 	}
     }
 
+    if (buf != NULL)
+        free(buf);
```

Comment:
`free(NULL);` is a no-op on any non-broken C library implementation.
– C99 standard, section 7.20.3.2/2